coremark/port/core_portme.h.in

```diff
@@ -0,0 +1,191 @@
+/*
+ * SPDX-FileCopyrightText: 2018 EEMBC
+ * SPDX-FileContributor: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Topic : Description
+	This file contains configuration constants required to execute on different platforms
+*/
+#ifndef CORE_PORTME_H
+#define CORE_PORTME_H
+/************************/
+/* Data types and settings */
+/************************/
+
+#include <stdint.h>
+#include "esp_idf_version.h"
+
+/* Configuration : HAS_FLOAT 
+	Define to 1 if the platform supports floating point.
+*/
+#ifndef HAS_FLOAT 
+#define HAS_FLOAT 1
+#endif
+/* Configuration : HAS_TIME_H
+	Define to 1 if platform has the time.h header file,
+	and implementation of functions thereof.
+*/
+#ifndef HAS_TIME_H
+#define HAS_TIME_H 0
+#endif
+/* Configuration : USE_CLOCK
+	Define to 1 if platform has the time.h header file,
+	and implementation of functions thereof.
+*/
+#ifndef USE_CLOCK
+#define USE_CLOCK 0
+#endif
+/* Configuration : HAS_STDIO
+	Define to 1 if the platform has stdio.h.
+*/
+#ifndef HAS_STDIO
+#define HAS_STDIO 1
+#endif
+/* Configuration : HAS_PRINTF
+	Define to 1 if the platform has stdio.h and implements the printf function.
+*/
+#ifndef HAS_PRINTF
+#define HAS_PRINTF 1
+#endif
+
+/* Definitions : COMPILER_VERSION, COMPILER_FLAGS, MEM_LOCATION
+	Initialize these strings per platform
+*/
+#ifndef COMPILER_VERSION 
+ #ifdef __GNUC__
+ #define COMPILER_VERSION "GCC"__VERSION__
+ #else
+ #define COMPILER_VERSION "Please put compiler version here (e.g. gcc 4.1)"
+ #endif
+#endif
+#ifndef COMPILER_FLAGS 
+ #define COMPILER_FLAGS "$<JOIN:$<FILTER:$<GENEX_EVAL:$<TARGET_PROPERTY:COMPILER_OPT>>,EXCLUDE,^-(([DWI])|(fmacro)).*>, >"
+#endif
+#ifndef MEM_LOCATION 
+ #define MEM_LOCATION "IRAM"
+#endif
+
+/* Data Types :
+	To avoid compiler issues, define the data types that need ot be used for 8b, 16b and 32b in <core_portme.h>.
+	
+	*Imprtant* :
+	ee_ptr_int needs to be the data type used to hold pointers, otherwise coremark may fail!!!
+*/
+typedef int16_t ee_s16;
+typedef uint16_t ee_u16;
+typedef int ee_s32;
+typedef double ee_f32;
+typedef uint8_t ee_u8;
+typedef unsigned long ee_u32;
+typedef ee_u32 ee_ptr_int;
+
+#define ee_size_t       size_t
```

Comment:
Probably no specific reason, it could be a typedef as well... This file is mostly just a copy of core_portme.h from coremark repo, I have only done minimal changes here.